packages/block-library/src/template-part/edit/utils/hooks.js

```diff
@@ -148,7 +149,9 @@ export function useTemplatePartArea( area ) {
 			);
 
 			return {
-				icon: selectedArea?.icon || defaultArea?.icon,
+				icon: getTemplatePartIcon(
```

Comment:
I wonder if this is better outside useSelect (although it's a fixed value, so no real issues as is)

Reply:
Do you mean using `getTemplatePartIcon` here?

https://github.com/WordPress/gutenberg/blob/95108ddbf24f6f67533de27affeb4e91a306f6ff/packages/block-library/src/template-part/edit/placeholder.js#L56-L57

Reply:
Yes, it should be fine for the current use case. Selectors like `getPostIcon` also use this helper and similar icon maps internally.